stdlib/src/base64/_b64encode.mojo

```diff
@@ -0,0 +1,408 @@
+    DType.uint8, simd_width
+]:
+    alias mask_1 = _repeat_until[simd_width](
+        SIMD[DType.uint8, 4](0b11111100, 0, 0, 0)
+    )
+    var masked_1 = shuffled_vector & mask_1
+    var result = masked_1 >> 2
+    return result
+
+
+fn _move_second_group_of_6_bits[
+    simd_width: Int
+](shuffled_vector: SIMD[DType.uint8, simd_width]) -> SIMD[
+    DType.uint8, simd_width
+]:
+    alias mask_2 = _repeat_until[simd_width](
+        SIMD[DType.uint8, 4](0b00000011, 0b11110000, 0, 0)
+    )
+    var masked_2 = shuffled_vector & mask_2
+    var masked_2_as_uint16 = bitcast[DType.uint16, simd_width // 2](masked_2)
+    var rotated_2 = bit.rotate_bits_right[4](masked_2_as_uint16)
+    var result = bitcast[DType.uint8, simd_width](rotated_2)
+    return result
+
+
+fn _move_third_group_of_6_bits[
+    simd_width: Int
+](shuffled_vector: SIMD[DType.uint8, simd_width]) -> SIMD[
+    DType.uint8, simd_width
+]:
+    alias mask_3 = _repeat_until[simd_width](
+        SIMD[DType.uint8, 4](
+            0,
+            0,
+            0b00001111,
+            0b11000000,
+        )
+    )
+    var masked_3 = shuffled_vector & mask_3
+    var masked_3_as_uint16 = bitcast[DType.uint16, simd_width // 2](masked_3)
+    var rotated_3 = bit.rotate_bits_left[2](masked_3_as_uint16)
+    var result = bitcast[DType.uint8, simd_width](rotated_3)
+    return result
+
+
+fn _move_fourth_group_of_6_bits[
+    simd_width: Int
+](shuffled_vector: SIMD[DType.uint8, simd_width]) -> SIMD[
+    DType.uint8, simd_width
+]:
+    alias mask_4 = _repeat_until[simd_width](
+        SIMD[DType.uint8, 4](
+            0,
+            0,
+            0,
+            0b00111111,
+        )
+    )
+    result = shuffled_vector & mask_4
+    return result
+
+
+fn _shuffle_input_vector[
+    simd_width: Int
+](input_vector: SIMD[DType.uint8, simd_width]) -> SIMD[DType.uint8, simd_width]:
+    # We reorder the bytes to fall in their correct 4 bytes chunks
+    # When Mojo is a bit more flexible with compile-time programming, we should be
+    # able to make this less verbose.
+    @parameter
+    if simd_width < 4:
+        constrained[False, msg="simd_width must be at least 4"]()
+        return SIMD[DType.uint8, simd_width]()  # dummy, unreachable
+    elif simd_width == 4:
+        return input_vector.shuffle[0, 1, 1, 2]()
+    elif simd_width == 8:
+        return input_vector.shuffle[0, 1, 1, 2, 3, 4, 4, 5]()
+    elif simd_width == 16:
+        return input_vector.shuffle[
+            0, 1, 1, 2, 3, 4, 4, 5, 6, 7, 7, 8, 9, 10, 10, 11
+        ]()
+    elif simd_width == 32:
+        # fmt: off
+        return input_vector.shuffle[
+            0, 1, 1, 2, 
+            3, 4, 4, 5,
+            6, 7, 7, 8, 
+            9, 10, 10, 11, 
+            12, 13, 13, 14, 
+            15, 16, 16, 17, 
+            18, 19, 19, 20, 
+            21, 22, 22, 23
+        ]()
+        # fmt: on
+    elif simd_width == 64:
+        # fmt: off
+        return input_vector.shuffle[
```

Comment:
**Question** I think this is failing on Graviton (such as m7i that we test on internally):

```
mojo --debug-level full -D ASSERT=all /mnt/engflow/worker/work/0/exec/bazel-out/k8-opt-release/bin/open-source/mojo/stdlib/test/base64/test_base64.mojo.test.runfiles/_main/open-source/mojo/stdlib/test/base64/test_base64.mojo
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:113:4: error: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:96:4: note: function instantiation failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:108:57: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:68:4: note: function instantiation failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:84:18: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:105:8: note: function instantiation failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/base64/_b64encode.mojo:383:38: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/base64/_b64encode.mojo:218:4: note: function instantiation failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/base64/_b64encode.mojo:224:48: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/base64/_b64encode.mojo:155:4: note: function instantiation failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/base64/_b64encode.mojo:211:10: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/simd.mojo:1965:45: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/simd.mojo:1910:10: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/constrained.mojo:56:6: note: constraint failed: size of the mask must match the output SIMD size
mojo: error: failed to run the pass manager
```

Reply:
Thanks, I'll take a look. I wonder if we can add a similar cpu in github actions to catch those errors in the public CI.

Reply:
I think the fix should be straight forward, just remove the `63`.

Reply:
Thanks for the fix @soraros ! I pushed the change in the latest commit!

Reply:
When we set up the GH workflows for the OSS stdlib, we intentionally didn't want run on anything internal (e.g. using any of our infra that sits on top of AWS EC2 VMs like m7i, m7g, etc.).  This is both from a security perspective and a cost thing.  In the current state, we're just running basic stdlib unit tests on free GitHub-provided hosts.  This is "mostly sufficient" as we've seen rather than running every OSS PR on the flurry of hardware and things we test on internally.

Reply:
That's still not quite right: the input shuffled vector has way too many elements still (more than 64).  I think you want

```
-            48, 49, 49, 50,
-            51, 52, 52, 53,
-            54, 55, 55, 56,
-            57, 58, 58, 59,
-            60, 61, 61, 62,
```
as a diff which brings the input shuffled vector to contain 64 elements.  This passes on a m7i locally for me, for example.